### PR TITLE
feat(branches): add schema for validating shift_plan objects that are part of branch table

### PR DESCRIPTION
**Pull request by galiot-ifl**

### CODE REVIEW

**Comment by galiot-ifl** [comment]
@aleksey-a-maltsev

**Comment by galiot-ifl** [comment]
feat(branches): add schema for validating shift_plan objects that are part of branch table  (reopen i closed by mistake)

File: lib/v0/branches/index.js [hunk 1]
```diff
@@ -1,3 +1,7 @@
 module.exports.create = require('./create')
 module.exports.read = require('./read')
 module.exports.update = require('./update')
+module.exports.shift_plan = {
+  read: require('./shift_plan/read'),
+  update: require('./shift_plan/update')
+}
```
Comment (aleksey-a-maltsev) [inline lib/v0/branches/index.js:7]: Why is shift plan connected to branch? For me it's either independed entity or staff-related. Can employee have individual shift plan?

Comment (galiot-ifl) [reply lib/v0/branches/index.js:7]: Business logic requires us to do that i will try to explain, so first of all, branches have working days configuration where manager can set what days are work days and what are not working days (mostly weekends) also you can set what are opening hours and closing hours and when break starts and when break ends for each day. This lives in configurations table, same as all other configs that dashboard manages from "Settings". Than next to this rough configuration of stores, we have something that we internally call "Employee specific work hours" where each staff member of certain branch can decide what are his/hers work hours and override basic configuration things like move his personal break to one hour before/after time specified during branch configuration. But wait! There is one more way to override everything and that is using shift plans. In special cases Manager of the branch can decide to override everything(worktime, break time) for everyone for some time.

Comment (galiot-ifl) [reply lib/v0/branches/index.js:7]: shift_plan**s** are way of full control, where granularity is on a level of a day and on a level of a staff member.

Comment (galiot-ifl) [reply lib/v0/branches/index.js:7]: https://github.com/tillhub/cli/pull/186

Comment (aleksey-a-maltsev) [reply lib/v0/branches/index.js:7]: So if shift plan is staff member related I would better create the shift plan entity and describe it independetly (not in branch). And left in branch only field which track enabled-disabled flag for shift plans in the branch.

**Review by aleksey-a-maltsev — COMMENTED** [review]
So if shift plan is staff member related I would better create the shift plan entity and describe it independetly (not in branch). And left in branch only field which track enabled-disabled flag for shift plans in the branch.

@mohsen89z What do you think? Yet another JSONB will lead us to many usses when we want to analyse the shift plan.

**Review by aleksey-a-maltsev — APPROVED** [review]
Approved but better to define shift plan as top level entity.

**Pull request merged**